iso15118/shared/messages/din_spec/body.py

```diff
@@ -249,15 +250,19 @@ def validate_requested_energy_mode(cls, values):
             values.get("dc_ev_charge_parameter"),
         )
         if requested_energy_mode not in ("DC_extended", "DC_core"):
-            raise ValueError(
-                f"Wrong energy transfer mode transfer mode {requested_energy_mode}"
+            raise V2GMessageValidationError(
+                f"Wrong energy transfer mode transfer mode {requested_energy_mode}",
+                ResponseCode.FAILED_WRONG_ENERGY_TRANSFER_MODE,
+                None,
             )
         if ("AC_" in requested_energy_mode and dc_params) or (
             "DC_" in requested_energy_mode and ac_params
         ):
-            raise ValueError(
+            raise V2GMessageValidationError(
                 "Wrong charge parameters for requested energy "
-                f"transfer mode {requested_energy_mode}"
+                f"transfer mode {requested_energy_mode}",
+                ResponseCode.FAILED_WRONG_ENERGY_TRANSFER_MODE,
```

Comment:
here should be `FAILED_WRONG_CHARGE_PARAMETER`

Reply:
Done